docs/3-building-blocks-of-oop-part-2/hometask/src/Weapon.ts

```diff
@@ -0,0 +1,47 @@
+import { Item } from "./Item";
+
+export abstract class Weapon extends Item {
+  protected static readonly MODIFIER_CHANGE_RATE = 0.05;
+  protected damageModifier = 0;
+  protected durabilityModifier = 0;
+
+  constructor(
+    name: string,
+    protected baseDamage: number,
+    protected baseDurability: number,
+    value: number,
+    weight: number
+  ) {
+    super(name, value, weight);
+  }
+
+  use(): string {
+    if (this.baseDurability <= 0) {
+      return `You can't use the ${this.name}, it is broken.`;
+    }
+    this.baseDurability = this.baseDurability - Weapon.MODIFIER_CHANGE_RATE;
+    const weaponUsageMessage = `You use the ${this.name}, dealing ${Weapon.MODIFIER_CHANGE_RATE} points of damage.`;
```

Comment:
please, do not modify `this.baseDurability`, the `this.durabilityModifier` should be changed instead: `this.durabilityModifier -= Weapon.MODIFIER_CHANGE_RATE`